src/block-management/block-manager.js

```diff
@@ -115,12 +117,14 @@ export default class BlockManager extends React.Component<PropsType, StateType>
 	}
 
 	static getDerivedStateFromProps( props: PropsType, state: StateType ) {
-		if ( props.fullparse === true ) {
+		if ( state.blocks !== props.blocks ) {
```

Comment:
This might impact performance as the list of blocks will be recreated on any change. I'll see if we can do something about it.